### PR TITLE
usm: Change simplify protocols

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Simplifies protocols setup in USM.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/monitor.go [hunk 1]
```diff
@@ -160,33 +159,40 @@ func (m *Monitor) Start() error {
 		}
 	}()
 
+	// Deleting from an array while iterating it is not a simple task. Instead, every successfully enabled protocol,
+	// we'll keep it in a temporary copy, and in case of a mismatch (a.k.a., we have a failed protocols) between
+	// enabledProtocolsTmp to m.enabledProtocols, we'll use the enabledProtocolsTmp.
+	enabledProtocolsTmp := m.enabledProtocols[:0]
 	for protocolType, protocol := range m.enabledProtocols {
```
Comment (Yumasi) [inline pkg/network/usm/monitor.go:166]:
```suggestion
	for _, protocol := range m.enabledProtocols {
```

File: pkg/network/usm/monitor.go [hunk 2]
```diff
@@ -160,33 +159,40 @@ func (m *Monitor) Start() error {
...
 		startErr := protocol.PreStart(m.ebpfProgram.Manager.Manager)
 		if startErr != nil {
-			delete(m.enabledProtocols, protocolType)
 			log.Errorf("could not complete pre-start phase of %s monitoring: %s", protocolType, startErr)
 			continue
+		} else {
+			enabledProtocolsTmp = append(enabledProtocolsTmp, protocol)
 		}
```
Comment (Yumasi) [inline pkg/network/usm/monitor.go:173]:
```suggestion
			continue
		}
		enabledProtocolsTmp = append(enabledProtocolsTmp, protocol)
```

File: pkg/network/usm/monitor.go [hunk 3]
```diff
@@ -160,33 +159,40 @@ func (m *Monitor) Start() error {
...
+		m.enabledProtocols = enabledProtocolsTmp
 	}
 
 	err = m.ebpfProgram.Start()
 	if err != nil {
 		return err
 	}
 
+	enabledProtocolsTmp = m.enabledProtocols[:0]
 	for protocolType, protocol := range m.enabledProtocols {
```
Comment (Yumasi) [inline pkg/network/usm/monitor.go:189]:
```suggestion
	for _, protocol := range m.enabledProtocols {
```

**Review by Yumasi — APPROVED** [review]
A few minor things, but other than those it looks good to me!

File: pkg/network/usm/monitor_testutil.go [hunk 4]
```diff
@@ -11,10 +11,8 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/cilium/ebpf"
-	"github.com/stretchr/testify/require"
-
 	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
```
Comment (DanielLavie) [inline pkg/network/usm/monitor_testutil.go:15]: Fix order

**Review by DanielLavie — APPROVED** [review]

**Pull request merged**